jobs/build-charms/builder_local.py

```diff
@@ -540,7 +540,8 @@ def pull_layers(self):
         layers_to_pull = [
             layer_name
             for layer in self.layers
-            if (layer_name := next(iter(layer))) != "layer:index"
+            for layer_name, layer_ops in layer.items()
+            if layer_ops.get("build_cache") is not False
```

Comment:
dangerously clever here, i love it!

Reply:
If you think this is dangerous -- i'll be happy to do something more clear

Reply:
nah, all good -- i like it using this `build_cache` opt vs matching the layer name.